src/cript/nodes/primary_nodes/material.py

```diff
@@ -445,27 +445,24 @@ def property(self, new_property_list: List[Any]) -> None:
 
     @classmethod
     def _from_json(cls, json_dict: dict):
-        # TODO find a better way to know identifiers
-        all_identifier_list = [
-            "amino_acid",
-            "bigsmiles",
-            "cas",
-            "chem_formula",
-            "chem_repeat",
-            "chemical_id",
-            "inchi",
-            "inchi_key",
-            "lot_number",
-            "mol_form",
-            "names",
-            "pubchem_cid",
-            "smiles",
-            "vendor",
-        ]
+
+        from cript.api.api import _get_global_cached_api
+        api = _get_global_cached_api()
+
+        # get JSON vocabulary of all "material_identifier_key"
+        all_identifiers: List[dict] = api.get_vocab_by_category("material_identifier_key")
+
+        # placeholder for what eventually will be all the names of material identifiers
```

Comment:
The logic here is fine.
But the comment really confused me.
The variable is not really a placeholder.
It actually is the list of all string names of identifiers.
You fill them right away with a for loop.

If you need a comment to explain that, consider using list compaction and no comment.

Reply:
yeah I agree, it's a bit confusing and a bit messy so I honestly don't love it, I wasn't sure how to clean it up.

I could do something like this, but I feel like this is even more confusing

```python
        all_identifiers_list = [
            identifier.get("name")
            for identifier in api.get_vocab_by_category(cript.ControlledVocabularyCategories.MATERIAL_IDENTIFIER_KEY)
        ]
```